pycrunch/shoji.py

```diff
@@ -247,3 +267,29 @@ def graph(self, newgraph):
         """Update the Order with the new graph."""
         self['graph'] = newgraph
         super(Order, self).put(data=self.json)
+
+
+class TaskProgressTimeoutError(Exception):
+    def __init__(self, entity, response):
+        super(TaskProgressTimeoutError, self).__init__(
+            'Task Progress did not complete before timeout. '
+            'Trap this exception and call exc.entity.wait_progress(exc.response) '
```

Comment:
Fair enough, though the R one is settable as a global/local option (and most people I know of tend to set it high enough to never time out, which says something). I didn't see how this one could be set by a user.